scripts/prepare_heat_data.py

```diff
@@ -64,68 +125,211 @@ def prepare_heat_data(n):
     district_heat_share = nodal_energy_totals["district heat share"]  # .round(2)
     nodal_energy_totals = nodal_energy_totals.multiply(pop_layout.fraction, axis=0)
 
+    energy_col_names = nodal_energy_totals.columns
+
+    # TODO reimplement using DRY approach
+    # a proper structure of the inputs is essential
+    residential_cols = energy_col_names[energy_col_names.str.contains("residential")]
+    services_cols = energy_col_names[energy_col_names.str.contains("services")]
+
+    energy_residential = nodal_energy_totals.loc[:, residential_cols].sum().sum()
+    energy_services = nodal_energy_totals.loc[:, services_cols].sum().sum()
+    energy_total = nodal_energy_totals.sum().sum()
+
+    nodal_energy_totals[f"heat residential space"] = (
+        SHARE_HEAT_RESID_DEMAND * energy_residential
+    )
+    nodal_energy_totals[f"heat residential water"] = (
+        SHARE_WATER_RESID_DEMAND * energy_residential
+    )
+    nodal_energy_totals[f"heat services space"] = (
+        SHARE_HEAT_SERVICES_DEMAND * energy_services
+    )
+    nodal_energy_totals[f"heat services water"] = (
+        SHARE_WATER_SERVICES_DEMAND * energy_services
+    )
+
+    nodal_energy_totals[f"cool residential space"] = (
+        SHARE_COOL_RESID_DEMAND * energy_residential
+    )
+    nodal_energy_totals[f"cool services space"] = (
+        SHARE_COOL_SERVICES_DEMAND * energy_services
+    )
+
+    nodal_energy_totals[f"electricity residential space"] = (
+        SHARE_ELECTRICITY_RESID_SPACE * energy_residential
+    )
+    nodal_energy_totals[f"electricity services space"] = (
+        SHARE_ELECTRICITY_SERVICES_SPACE * energy_services
+    )
+
+    # TODO Add a local fix to make test with pre-compiled networks
+    # snapshots = n.snapshots
+
+    # heating/cooling demand profiles
     # copy forward the daily average heat demand into each hour, so it can be multiplied by the intraday profile
     daily_space_heat_demand = (
         xr.open_dataarray(snakemake.input.heat_demand_total)
         .to_pandas()
-        .reindex(index=n.snapshots, method="ffill")
+        .reindex(index=snapshots, method="ffill")
+    )
+    daily_space_cooling_demand = (
+        xr.open_dataarray(snakemake.input.cooling_demand_total)
+        .to_pandas()
+        .reindex(index=snapshots, method="ffill")
     )
 
-    intraday_profiles = pd.read_csv(
+    intraday_profiles_heating = pd.read_csv(
         snakemake.input.heat_profile, index_col=0
     )  # TODO GHALAT
+    intraday_profiles_cooling = pd.read_csv(
+        snakemake.input.cooling_profile, index_col=0
+    )  # TODO GHALAT
+
+    calibr_heat_df = pd.read_csv(os.path.join(CALIBR_DIR, CALIBR_HEAT_FL), index_col=0)
+    calibr_heat_df["country"] = "US"
+    calibr_cool_df = pd.read_csv(os.path.join(CALIBR_DIR, CALIBR_COOL_FL), index_col=0)
+    calibr_cool_df["country"] = "US"
+
+    calibr_heat_buses_df = locate_bus(
+        df=calibr_heat_df,
+        countries=["US"],
```

Comment:
These in-code "ad-hoc" solutions are better tagged with a TODO or avoided if possible; can't we load it from the snakemake parameters?